cuda/matrix/csr_kernels.cu

```diff
@@ -644,14 +648,17 @@ int compute_items_per_thread(std::shared_ptr<const CudaExecutor> exec)
     case 0x37:
         num_item = 14;
     }
```

Comment:
Just out of curiosity: Where did you get these numbers? Did you benchmark them, or did you derive them from some properties?
It looks like it does not directly correlate with the number of cuda-cores / SM. If it was, then the map `nGpuArchCoresPerSM` would be a better choice.

Reply:
I use the storage of shared_memory to make the occupancy 100%

Reply:
Ah, thanks, that makes sense.
And I guess CUDA does not have an easy way of accessing this information (like with many things).

Can you add a comment (pretty much what you just said) to the function or the `switch`-`case` block, including a `TODO` to extend this list when new cards are released (to achieve maximum performance)?

Reply:
Sure, I have done